prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -106,6 +109,31 @@ def map_lines_to_substations_using_coords(
         ]
     )
     tree = KDTree([ll2uv(p[1], p[0]) for p in subcoord])
+
+    # Find points which would get mapped beyond our max distance
+    tree_query_results = [(p, tree.query(ll2uv(p[1], p[0]))) for p in missing_points]
+    to_add_list = [
+        {
+            "LATITUDE": p[0],
+            "LONGITUDE": p[1],
+            "NAME": str(p),
+            "STATE": substations.loc[subcoord2subid[subcoord[closest_idx]][0], "STATE"],
+        }
+        for p, (dist, closest_idx) in tree_query_results
+        if dist > max_remap
+    ]
+    substations_to_add = pd.DataFrame(
+        to_add_list, index=pd.RangeIndex(len(to_add_list)) + substations.index.max() + 1
+    )
+    print(f"Adding {len(substations_to_add)} new substations for unmapped endpoints")
+    # Append the new ones, then re-generate the mapping
+    substations = pd.concat([substations, substations_to_add])
+    substations.index.name = "ID"
+    subcoord2subid = (
+        substations.round(rounding).groupby(["LATITUDE", "LONGITUDE"]).groups
+    )
+    subcoord = list(subcoord2subid)
+    tree = KDTree([ll2uv(p[1], p[0]) for p in subcoord])
```

Comment:
Maybe we can factor this out as an internal function to avoid duplicated lines.

Reply:
Done